presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/HashGenerationOptimizer.java

```diff
@@ -970,8 +959,8 @@ public VariableReferenceExpression getRequiredHashVariable(HashComputation hash)
     {
         Map<VariableReferenceExpression, VariableReferenceExpression> outputToInput = new HashMap<>();
         for (Map.Entry<VariableReferenceExpression, RowExpression> assignment : assignments.entrySet()) {
-            if (castToExpression(assignment.getValue()) instanceof SymbolReference) {
-                outputToInput.put(assignment.getKey(), new VariableReferenceExpression(((SymbolReference) castToExpression(assignment.getValue())).getName(), assignment.getKey().getType()));
+            if (assignment.getValue() instanceof VariableReferenceExpression) {
```

Comment:
Is it helpful to verify that `assignment.getValue()` is never an `OriginalExpression` here?

Reply:
Usually, if it is still `OriginalExpression` we should be able to catch in the `ValidateNoOriginalExpression` rule @kewang1024 is adding in https://github.com/prestodb/presto/pull/13007, but it does not hurt to check though.

Reply:
That's not true. If the value is `OriginalExpression` this if statement will not execute, so you won't add that to `outputToInput`. It's not related to the plan sanity checker. Granted if that happens probably other stuff would fail, but debugging is way easier when a `checkArgument` fails than when something fails in a later stage.